fip-6.md

```diff
@@ -0,0 +1,141 @@
+---
+fip: 6
+title: Provide ability to deactivate FIO Addresses and Domains
+status: Draft
+type: Functionality
+author: Casey Gardiner <casey@dapix.io>
+created: 2020-04-22
+updated: 
+---
+
+## Abstract
+This FIP implements the following:
+* Adds the ability for FIO Address owners the ability to deactivate stale or unused addresses.
+* Adds the ability for FIO Domain owners the ability to set their domain as expired.
+* Adds new API end points for FIO Domain deactivation and FIO Address burn. 
+* Adds new fees for FIO Domain deactivation and FIO Address burn.
+
+## Motivation
+Presently the FIO Blockchain only burns and removes old addresses and domains once they reach their expiration date. Added functionality to remove these addresses and domains at any given time gives the owners flexibility.
+
+## Specification
+### Burn Addresses
+#### New end point: *deactivate_fio_address* 
+#### New action in new fio.address contract burnaddress
+##### Request
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|fio_address|Yes|FIO Address, see FIO Address validation rules.|FIO Address that is being requested to burn.|
+|actor|Yes|FIO account name|FIO account for the signer, the account owning this FIO Address.|
+|max_fee|Yes|max fee SUFs|Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by /get_fee for correct value.|
+|tpid|Yes|FIO Address of TPID, See FIO Address validation rules|FIO Address of the wallet which generates this transaction. This FIO Address will be paid 10% of the fee.See FIO Protocol#TPIDs for details. Set to empty if not known.|
+
+###### Example
+```
+{
+    "fio_address": "purse@alice",
+    "actor": "aftyershcu22",
+    "max_fee": 400000000,
+    "tpid": "rewards@wallet"
+}
+```
+##### Processing
+* Require auth of the actor
+* Request is validated per Exception handling.
+* Removes the fio_address from the fionames table.
+* Removes the TPID entry inside the tpid index table. 
+* Return status json containing status (ok), and fee charged.
+##### Response
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|Invalid FIO Address format|FIO Address format is not valid|400|"fio_address"|Value sent in, e.g. "purse@alice"|"Invalid FIO Address"|
+|Not owner of FIO Address|The signer does not own the address|403||||
+|Invalid fee value|max_fee format is not valid|400|"max_fee"|Value sent in, e.g. "-100"|"Invalid fee value"|
+|Insufficient funds to cover fee|Account does not have enough funds to cover fee|400|"max_fee"|Value sent in, e.g. "400000000"|"Insufficient funds to cover fee"|
+|Fee exceeds maximum|Actual fee is greater than supplied max_fee|400|max_fee"|Value sent in, e.g. "400000000"|"Fee exceeds supplied maximum"|
+|Invalid TPID|TPID is not empty or contains invalid FIO address|400|"tpid"|Value sent in is not empty and not a valid FIO Address format|"TPID must be empty or valid FIO address"|
```

Comment:
`empty or contains` should probably be `empty and contains`

Reply:
"TPID must be empty or valid FIO address" is the standard error across all associated endpoints. In software, something cannot be empty AND contain data.

Reply:
I was using a reference from Eds FIPs but I've changed it to match the current  1.0 API spec